packages/@sanity/codegen/src/typescript/typeGenerator.ts

```diff
@@ -82,16 +87,19 @@ export class TypeGenerator {
    * @internal
    * @beta
    */
-  generateTypeNodeTypes(identifierName: string, typeNode: TypeNode): string {
+  generateTypeNodeTypes(
+    identifierName: string,
+    typeNode: TypeNode,
+  ): {type: string; typeName: string} {
```

Comment:
To reuse the types later (like when generating the type map), we need to be able to reference them and for that we need the actual type name that was generated.

This change makes `generateTypeNodeTypes` return the generated names so we can reuse them.

Reply:
Hm it would be good not to break this method.

A few alternatives:
 - the `generateQueryMap` could also just use this private map to lookup the generated type name.
 - add a `addToQueryMap(name: string, type: string)` that collects all the queries generated in the class, then `generateQueryMap()` can just traverse that list and spit out the types for all.
- Expose the typeNameMap in a method, then we can extract the generated typeName separately.